credential/base/src/main/java/org/wildfly/security/password/spec/HashPasswordSpec.java

```diff
@@ -52,7 +52,7 @@ public byte[] getDigest() {
 
     @Override
     public boolean equals(Object other) {
-        return other instanceof HashPasswordSpec && Arrays.equals(digest, ((HashPasswordSpec)other).digest);
+        return other instanceof HashPasswordSpec && MessageDigest.equals(digest, ((HashPasswordSpec)other).digest);
```

Comment:
Just a small thing, this should be `MessageDigest.isEqual` instead of `MessageDigest.equals`.

Reply:
And then `import java.security.MessageDigest;` should be added before the `java.util` import.